src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+
+import {t} from 'app/locale';
+import MenuItem from 'app/components/menuItem';
+import DropdownLink from 'app/components/dropdownLink';
+import QueryCount from 'app/components/queryCount';
+import space from 'app/styles/space';
+
+export default class OrganizationSavedSearchSelector extends React.Component {
+  static propTypes = {
+    savedSearchList: PropTypes.array.isRequired,
+    onSavedSearchSelect: PropTypes.func.isRequired,
+    query: PropTypes.string,
+    queryCount: PropTypes.number,
+    queryMaxCount: PropTypes.number,
+    searchId: PropTypes.string,
+  };
+
+  getTitle() {
+    const {searchId, query, savedSearchList} = this.props;
+    let result;
+
+    if (searchId) {
+      result = savedSearchList.find(search => searchId === search.id);
+    } else {
+      result = savedSearchList.find(search => query === search.query);
+    }
+
+    return result ? result.name : t('Custom Search');
+  }
+
+  renderList() {
+    const {savedSearchList, onSavedSearchSelect} = this.props;
+
+    if (savedSearchList.length === 0) {
+      return <EmptyItem>{t("There don't seem to be any saved searches yet.")}</EmptyItem>;
+    }
+
+    return savedSearchList.map(search => (
+      <StyledMenuItem onSelect={() => onSavedSearchSelect(search)} key={search.id}>
+        <span>
+          <strong>{search.name}</strong>
+        </span>
+        <code>{search.query}</code>
```

Comment:
Will showing pinned state come later?

Reply:
It will - just need to get hold of an icon for this first